common/src/main/java/com/hedera/block/common/utils/FileUtils.java

```diff
@@ -0,0 +1,130 @@
+     * @throws IOException if the path cannot be created
+     */
+    public static void createPathIfNotExists(
+            @NonNull final Path toCreate,
+            @NonNull final System.Logger.Level logLevel,
+            @NonNull final FileAttribute<Set<PosixFilePermission>> perms,
+            @NonNull final String semanticPathName,
+            final boolean createDir)
+            throws IOException {
+        Objects.requireNonNull(toCreate);
+        Objects.requireNonNull(logLevel);
+        Objects.requireNonNull(perms);
+        StringUtils.requireNotBlank(semanticPathName);
+        final String type = createDir ? "directory" : "file";
+        if (Files.notExists(toCreate)) {
+            if (createDir) {
+                Files.createDirectories(toCreate, perms);
+            } else {
+                Files.createFile(toCreate, perms);
+            }
+            LOGGER.log(logLevel, "Created " + type + " [" + semanticPathName + "] at:" + toCreate);
+        } else {
+            LOGGER.log(
+                    logLevel,
+                    "Using existing " + type + " [" + semanticPathName + "] at:" + toCreate);
+        }
+    }
+
+    /**
+     * Read a GZIP file and return the content as a byte array.
+     *
+     * @param filePath Path to the GZIP file
+     * @return byte array of the content of the GZIP file
+     * @throws IOException if an I/O error occurs
+     */
+    public static byte[] readGzFile(@NonNull final Path filePath) throws IOException {
+        try (final InputStream fileInputStream =
+                        Files.newInputStream(Objects.requireNonNull(filePath));
+                final GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream)) {
+            return gzipInputStream.readAllBytes();
+        }
+    }
+
+    // TODO the below code required the .gz and .blk file extension strings that we agreed to remove
+    // please remove the method if you think it should not exist here, if it should, then should the
+    // string literals .gz and .blk be back in the Strings constants class, or introduced here
+    // directly
+    //    /**
+    //     * Read a file and return the content as a byte array.
+    //     *
+    //     * @param filePath Path to the file
+    //     * @return byte array of the content of the file or null if the file extension is not
+    // supported
+    //     * @throws IOException if an I/O error occurs
+    //     */
+    //    public static byte[] readFileBytes(@NonNull final Path filePath) throws IOException {
+    //        final String filePathAsString = Objects.requireNonNull(filePath).toString();
+    //        if (filePathAsString.endsWith(Strings.GZ_FILE_EXTENSION)) {
+    //            return readGzFile(filePath);
+    //        } else if (filePathAsString.endsWith(Strings.BLOCK_FILE_EXTENSION)) {
+    //            return Files.readAllBytes(filePath);
+    //        } else {
+    //            return null;
+    //        }
+    //    }
```

Comment:
Please resolve `TODO`

Reply:
Resolved by overloading the method to optionally accept the extensions as parameters.